### PR TITLE
[spirv] Add SM 6.6 8-bit packed types and intrinsics

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp [hunk 1]
```diff
@@ -9812,6 +9826,164 @@ SpirvEmitter::processIntrinsicLog10(const CallExpr *callExpr) {
...
+  // The result is four 8-bit values (32 bits in total) which are packed in an
+  // unsigned uint32_t.
+  //
+  // Note: uint8_t4_packed and int8_t4_packed are NOT vector types! They are
+  // both scalar 32-bit unsigned integer types where each byte represents one
+  // value.
+  //
+  // Note: In pack_clamp_{s|u}8 intrinsics, an input of 0x100 will be turned
+  // into 0xFF, not 0x00. Therefore, it is important to perform a clamp first,
+  // and then a truncation.
```
Comment (jaebaek) [inline tools/clang/lib/SPIRV/SpirvEmitter.cpp:9871]: +1. I like your description here.

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp [hunk 2]
```diff
@@ -9812,6 +9826,164 @@ SpirvEmitter::processIntrinsicLog10(const CallExpr *callExpr) {
...
+    auto *minInstr = spvBuilder.getConstantInt(
+        elemType, llvm::APInt(bitwidth, clampMin, isSigned));
+    auto *maxInstr = spvBuilder.getConstantInt(
+        elemType, llvm::APInt(bitwidth, clampMax, isSigned));
+    auto *minVec = spvBuilder.getConstantComposite(
+        argType, {minInstr, minInstr, minInstr, minInstr});
+    auto *maxVec = spvBuilder.getConstantComposite(
+        argType, {maxInstr, maxInstr, maxInstr, maxInstr});
+    auto clampOp = isSigned ? GLSLstd450SClamp : GLSLstd450UClamp;
+    // ehsan
```
Comment (jaebaek) [inline tools/clang/lib/SPIRV/SpirvEmitter.cpp:9902]: Remove `// ehsan`?

**Review by jaebaek — APPROVED** [review]
The code is well-written and easy to understand the context. Thanks!

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp (hunk 2 above)
Comment (ehsannas) [reply tools/clang/lib/SPIRV/SpirvEmitter.cpp:9902]: Ouch! 😄  how did this sneak in! Good catch. Thanks

**Pull request merged**